### PR TITLE
Don't ignore unbound static methods.

### PR DESCRIPTION
**Pull request by pmconne**
Prevent [mistakes like this](https://github.com/iTwin/itwinjs-core/pull/5409).

### CODE REVIEW

**Review by MichaelBelousov — APPROVED** [review]

File: dist/configs/itwinjs-recommended.js [hunk 1]
```diff
@@ -228,7 +228,7 @@ module.exports = {
     "@typescript-eslint/unbound-method": [
       "error",
       {
-        "ignoreStatic": true
+        "ignorestatic": false
```
Comment (aruniverse) [inline dist/configs/itwinjs-recommended.js:231]:
```suggestion
        "ignoreStatic": false
```

https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/unbound-method.md#ignorestatic

**Review by ben-polinsky — APPROVED** [review]

**Review by aruniverse — APPROVED** [review]
i _think_ you can delete the comment on L227

File: dist/configs/itwinjs-recommended.js (hunk 1 above)
Comment (pmconne) [reply dist/configs/itwinjs-recommended.js:231]: I hate laptop keyboards.

Comment (pmconne) [reply dist/configs/itwinjs-recommended.js:231]: Do you want me to submit another PR?

Comment (aruniverse) [reply dist/configs/itwinjs-recommended.js:231]: dw, ill do it

**Pull request merged**